pandas/core/indexes/multi.py

```diff
@@ -3095,8 +3103,54 @@ def _update_indexer(idxr, indexer=indexer):
         # empty indexer
         if indexer is None:
             return Int64Index([])._ndarray_values
+
+        if need_sort:
+            indexer = self._reorder_indexer(seq, indexer)
+
         return indexer._ndarray_values
 
+    def _reorder_indexer(self, seq, indexer: Int64Index) -> Int64Index :
+        """
+        Reorder an indexer of a MultiIndex (self) so that the label are in the
+        same order as given in seq
+
+        Parameters
+        ----------
+        seq : label/slice/list/mask or a sequence of such
+        indexer: an Int64Index for element of self
+
+        Returns
+        -------
+        indexer : a sorted Int64Index of element of self ordered as seq
```

Comment:
I don't understand "of element of self"

Reply:
I mean an indexer of self. I will rephrase it.